src/bodywork/workflow_execution.py

```diff
@@ -97,6 +97,12 @@ def run_workflow(
         env_vars = k8s.create_k8s_environment_variables(
             [(GIT_COMMIT_HASH_K8S_ENV_VAR, get_git_commit_hash(cloned_repo_dir))]
         )
+        if config.project.secrets_group:
+            _log.info(
+                f"Replicating {config.project.secrets_group} secrets in "
+                f"namespace={namespace}"
+            )
+            k8s.replicate_secrets_in_namespace(namespace, config.project.secrets_group)
```

Comment:
Should we explicitly handle the potential exceptions from this block, so that the log message are clear about what has gone wrong? Otherwise, we could be feeding the user a k8s API error payload, which is somewhat verbose.

Reply:
I agree, a one liner explaining what went wrong would make it easier for the user to understand at what point things failed. I'll extract the replicating of the secrets into a separate method as well to try and keep things neat and succinct in the `run_workflow` method.